lily4178993/maths-magicians
# Implement Math Logic and Button Functionality

# Implement Math Logic and Button Functionality

## Description
This pull request includes the implementation of math logic and button functionality for the Calculator component. The following tasks have been completed:

- Created a new directory called `logic` within the project.
- Copied the files `calculate.js` and `operate.js` into the `logic/` directory.
- Installed [big.js](https://www.npmjs.com/package/big.js) using NPM to handle precise arithmetic calculations.
- Analyzed the `calculate.js` and `operate.js` files to understand their functionality.
- Imported `calculate.js` into the `Calculator.js` component.
- Implemented click event handlers in the `Calculator` component using the math logic from `calculate.js` and `operate.js`.
- Ensured that all calculator buttons work according to their functionality (e.g., subtraction, addition, etc.).
- Tested the application in the browser to verify that the `Calculator` can perform math operations.

## Checklist
- [x] Created `logic` directory.
- [x] Copied `calculate.js` and `operate.js` to `logic/`.
- [x] Installed `big.js` dependency.
- [x] Imported `calculate.js` into `Calculator.js`.
- [x] Implemented click event handlers with math logic.
- [x] Tested the Calculator in the browser.

## Reviewer Notes
Please review the implementation of math logic and button functionality to ensure that the Calculator component can perform accurate math operations. Verify that all buttons are functioning as expected.

## How to Test
To test the Calculator component with the new math logic and button functionality:

1. Ensure you have cloned this branch and installed dependencies.
2. Run the development server:
   ```bash
   npm start
   ```
3. Open your web browser and access the application to interact with the Calculator.
